server/services/contentProperty.service.js

```diff
@@ -74,3 +75,29 @@ export const getHouseTypesByAreaId = async (areaId) => {
   }
   return ContentProperty.find({ areaId }).distinct('houseType');
 };
+
+export const getAllProperties = async (page = 1, limit = 10) => {
+  const properties = await ContentProperty.aggregate([
+    {
+      $lookup: {
+        from: 'areas',
+        localField: 'areaId',
+        foreignField: '_id',
+        as: 'areaInfo',
+      },
+    },
+    {
+      $unwind: '$areaInfo',
+    },
+    {
+      $facet: {
+        metadata: [{ $count: 'total' }, { $addFields: { page, limit } }],
+        data: [{ $skip: parseInt((page - 1) * limit, 10) }, { $limit: parseInt(limit, 10) }],
```

Comment:
Let's create a custom method that returns the array for data too to avoid repetition:

```suggestion
        data: generateFacetData(page, limit),
```

In `../helpers/pagination`, you will have something similar to:

```js
const generateFacetData = (page, limit) => [
    { $skip: parseInt((page - 1) * limit, 10) }, 
    { $limit: parseInt(limit, 10) }
];
```